### PR TITLE
Qubes ebuilds bump, again

### PR DESCRIPTION
**Pull request by pzskc383**

### CODE REVIEW

File: qubes-vm/qmemman/qmemman-4.0.22.ebuild [hunk 1]
```diff
@@ -40,4 +40,5 @@ install_systemd_units() {
 src_install() {
 	dosbin meminfo-writer
 	install_systemd_units "${SYSTEMD_UNITS[@]}"
+	newinitd "${FILESDIR}"/qubes-meminfo-writer.initd qubes-meminfo-writer
```
Comment (aidecoe) [inline qubes-vm/qmemman/qmemman-4.0.22.ebuild]: let's copy function from core-agent and use that for consistency. Later we will go through common functionality and put that together in an eclass or sth.

File: qubes-vm/qmemman/qmemman-4.0.25.ebuild [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+	for unit in "${@}"; do
+		systemd_dounit "${unit}"
+	done
+
+}
+
+src_install() {
+	dosbin meminfo-writer
+	install_systemd_units "${SYSTEMD_UNITS[@]}"
+	newinitd "${FILESDIR}"/qubes-meminfo-writer.initd qubes-meminfo-writer
```
Comment (aidecoe) [inline qubes-vm/qmemman/qmemman-4.0.25.ebuild]: let's copy function from core-agent and use that for consistency. Later we will go through common functionality and put that together in an eclass or sth.

File: qubes-vm/qrexec/qrexec-4.0.39.ebuild [hunk 3]
```diff
@@ -51,6 +48,7 @@ src_install() {
 	default
 	dotmpfiles "${FILESDIR}"/qubes-vm-qrexec.conf
 	install_systemd_units "${SYSTEMD_UNITS[@]}"
+	newinitd vm-init.d/qubes-qrexec-agent
```
Comment (aidecoe) [inline qubes-vm/qrexec/qrexec-4.0.39.ebuild]: let's copy function from core-agent and use that for consistency. Later we will go through common functionality and put that together in an eclass or sth.

Comment (aidecoe) [reply qubes-vm/qrexec/qrexec-4.0.39.ebuild]: isn't this missing second arg? or should it actually be `doinitd`?

File: qubes-vm/qrexec/qrexec-4.0.47.ebuild [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+src_compile() {
+	BACKEND_VMM=xen default
+}
+
+src_install() {
+	default
+	dotmpfiles "${FILESDIR}"/qubes-vm-qrexec.conf
+	install_systemd_units "${SYSTEMD_UNITS[@]}"
+	doinitd ../vm-init.d/qubes-qrexec-agent
```
Comment (aidecoe) [inline qubes-vm/qrexec/qrexec-4.0.47.ebuild]: let's copy function from core-agent and use that for consistency. Later we will go through common functionality and put that together in an eclass or sth.

**Review by aidecoe — CHANGES_REQUESTED** [review]

File: qubes-vm/db/files/no-systemd-deps.patch [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
++/* #include <systemd/sd-daemon.h> */
+ #else // !WIN32
+ // parameters for a client pipe thread
+ struct thread_param {
+@@ -788,7 +788,7 @@
+     /* now ready for serving requests, notify parent */
+     /* FIXME: OS dependent code */
+     if (getenv("NOTIFY_SOCKET")) {
+-        sd_notify(1, "READY=1");
++        // sd_notify(1, "READY=1");
```
[the remark this answers is not in the document]
Comment (marmarek) [reply qubes-vm/db/files/no-systemd-deps.patch]: There is more flexible approach at https://github.com/QubesOS/qubes-core-qubesdb/pull/8, but unfortunately it was never finished by the author.

File: qubes-vm/gui-agent/gui-agent-4.0.24.ebuild [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+	sed -e /postlogin/d \
+		-i ../appvm-scripts/etc/pam.d/qubes-gui-agent
+}
+
+src_compile() {
+	BACKEND_VMM=xen default
+	(
+		cd ../gui-common
+		BACKEND_VMM=xen default
+	)
```
[the remark this answers is not in the document]
Comment (aidecoe) [reply qubes-vm/gui-agent/gui-agent-4.0.24.ebuild]: gui-common package has something different. This is actually needed. Confusing.

**Comment by aidecoe** [comment]
I have picked the changes manually and either fixed or skipped where I was not sure.

**Pull request closed without merging**